doc/user/content/sql/types/list.md

```diff
@@ -0,0 +1,490 @@
+
+Detail | Info
+-------|------
+**Quick Syntax** | `LIST[[1,2],[3]]`
+**Size** | Variable
+
+## Syntax
+
+{{< diagram "type-list.svg" >}}
+
+Field | Use
+------|-----
+_element_ | An element of any [data type](../) to place in the list.
+
+## List functions + operators
+
+#### Polymorphism
+
+List functions and operators are polymorphic, which applies the following constraints to their
+arguments:
+
+- All instances of `listany` must be lists of the same type.
+- All instances of `listelementany` must be of the same type.
+  - If a function uses both `listany` and `listelementany`, all instances of
+    `listany` must be a list of the type used in `listelementany`.
+
+### Operators
+
+{{% list-operators %}}
+
+### Functions
+
+{{< fnlist "List" >}}
+
+## Details
+
+### Type name
+
+The name of a list type is the name of its element type followed by `list`, e.g.
+`int list`. This rule can be applied recursively, e.g. `int list list` for a
+`list` of `int list`s, i.e. a 2D list.
+
+### Construction
+
+You can construct lists using the `LIST` expression:
+
+```sql
+SELECT LIST[1, 2, 3];
+```
+```nofmt
+  list
+---------
+ {1,2,3}
+```
+
+You can nest `LIST` constructors to create multidimensional lists:
+
+```sql
+SELECT LIST[LIST['a', 'b'], LIST['c']];
+```
+```nofmt
+    list
+-------------
+ {{a,b},{c}}
+```
+
+You can also elide the `LIST` keyword from the interior list expressions:
+
+```sql
+SELECT LIST[['a', 'b'], ['c']];
+```
+```nofmt
+    list
+-------------
+ {{a,b},{c}}
+```
+
+Multidimensional lists can be "ragged", i.e. the length of each nested list
+can differ from others. This differs from `array`, which does not
+support ragged multideimensional arrays, i.e. the length of each nested array
+must be the same.
+
+Note that you can also construct lists using the available [`text`
+cast](#text-to-list-casts).
+
+### Accessing lists
+
+#### Elements (subscripting)
+
+Accessing elements of `list`s (known as subscripting) uses brackets (`[]`)
+and 1-index element positions:
+
+```sql
+SELECT LIST[['a', 'b'], ['c']][1];
+```
+```nofmt
+ ?column?
+----------
+ {a,b}
+```
+
+Subscripts can be chained together to descend the list's dimensions:
+
+```sql
+SELECT LIST[['a', 'b'], ['c']][1][2];
+```
+```nofmt
+ ?column?
+----------
+ b
+```
+
+If the index is invalid (either less than 1, or greater the maximum index),
+lists return _null_.
+
+```sql
+SELECT LIST[['a', 'b'], ['c']][1][5] AS exceed_index;
+```
+```nofmt
+ exceed_index
+--------------
+
+```
+
+Lists have types based on their dimension (unlike arrays), and error if you
+attmept to subscript a non-list element (i.e. the depth of subscripting exceeds
+the list'd maximum dimension):
+
+```sql
+SELECT LIST[['a', 'b'], ['c']][1][2][3];
+```
+```nofmt
+ERROR:  cannot subscript type string
+```
+
+#### Ranges/slices (subscripting)
+
+Slicing `lists` (confusingly _also_ known as subscripting) uses the format
```

Comment:
It's general database terminology; I wanted to leave it in the heading so it would come up in searches, but we'll see if removing it from the headings negatively impacts the page's findability.